src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -150,9 +150,131 @@ begin
   simp [e₂, direct_sum.submodule_coe, direct_sum.to_module, dfinsupp.sum_add_hom_apply]
 end
 
+variables (ι) (𝕜) (E) [fintype ι] [decidable_eq ι]
+
+
+def euclidean_space.single {𝕜 : Type*} {ι : Type*} [fintype ι] [is_R_or_C 𝕜] (i : ι) (a : 𝕜) :
+ euclidean_space 𝕜 ι :=
+  set.indicator {i} (λ j, a)
+
+
+theorem euclidean_space.single_apply {𝕜 : Type*} {ι : Type*} [fintype ι] [decidable_eq ι]
```

Comment:
```suggestion
@[simp] theorem euclidean_space.single_apply {𝕜 : Type*} {ι : Type*} [fintype ι] [decidable_eq ι]
```
Tagging this with `simp` will simplify later proofs (for example, the next one).